pkg/server/server.go

```diff
@@ -25,13 +27,48 @@ const (
 type Server struct {
 	bucket objstore.Bucket
 	logger log.Logger
+
+	validations          *prometheus.CounterVec
+	validationFailures   *prometheus.CounterVec
+	ruleGroupsConfigured *prometheus.GaugeVec
+	rulesConfigured      *prometheus.GaugeVec
 }
 
-func NewServer(bucket objstore.Bucket, logger log.Logger) *Server {
-	return &Server{
+func (s *Server) registerServerMetrics(reg *prometheus.Registry) {
+	//nolint:exhaustivestruct
```

Comment:
why not just have this in `NewServer` directly? It's never called anywhere else and it's pretty common practice to init metrics in the constructor.

Reply:
Sure! Done in https://github.com/observatorium/rules-objstore/pull/9/commits/55d01f7ab7d75c7bffd9d04247c3102ddccd8121